cli/flags.rs

```diff
@@ -56,6 +56,8 @@ pub enum DenoSubcommand {
   Lint {
     files: Vec<String>,
     rules: bool,
+    json: bool,
+    json_sort: bool,
```

Comment:
**What's up with `json_sort`?**

This was needed to enforce determinism with the integration spec. `json_sort` is a flag sent to the `JsonLintReporter` to sort the lints by location before serialization.

Reply:
I am for just always sorting (maybe even in the pretty printer too?). @bartlomieju WDYT?

Reply:
Agree, `--sort` is superfluous

Reply:
> I am for just always sorting (maybe even in the pretty printer too?)

Sounds good to me! Since JSON has to wait tell the whole object is ready anyways, the cost of just always sorting is fairly small :smile:

IMHO, we shouldn't sort in the `PrettyLintReporter`. If we wanted to do this we'd either need to store all diagnostics and wait to `eprintln` tell the very end, or introduce some complexity such that we're guaranteed that the file's are visited and resolved in a deterministic order. For the pretty printer, I'm not sure sorting is worth it 🤔

It's really nice that the default pretty reporter is responsive (i.e., it outputs results right away) and has very low overhead (i.e., it isn't burdened with remembering diagnostics). Most use cases I can think of for wanting to run `deno lint`, I think responsiveness and low overhead are most important. 🤔

Maybe we should create a follow-up issue about sorting the result of the `PrettyLintReporter` and continue the discussion there?